django/core/handlers/middleware.py

```diff
@@ -0,0 +1,10 @@
+class MiddlewareMixin(object):
+    def __call__(self, request, response_factory):
+        response = None
+        if hasattr(self, 'process_request'):
+            response = self.process_request(request)
+        if not response:
+            response = response_factory(request)
```

Comment:
Yes, you're right, I see that for purposes of backwards-compatibility with old-style middleware (which is the point of this mix-in), this is adequate as-is.